Marketplace/voiceHUDApp/appResources/appData/voiceHUD_app.js

```diff
@@ -0,0 +1,836 @@
+                        x: 0.0631,
+                        y: 0.003,
+                        z: 0
+                    },
+                    dimensions: HUD_DEFAULT_DIMENSIONS,
+                    alpha: BAR_ALPHA,
+                    localRotation: {
+                        x: 0,
+                        y: 0,
+                        z: 0,
+                        w: 1
+                    },
+                    solid: true,
+                    ignoreForCollisions: true,
+                    color: {
+                        "red": 0,
+                        "green": 255,
+                        "blue": 255
+                    },
+                    parentID: user.hudID[0]
+                },
+                {
+                    name: "Voice-HUD",
+                    drawInFront: true,
+                    shape: "Quad",
+                    localPosition: {
+                        x: 0.1217,
+                        y: 0.003,
+                        z: 0
+                    },
+                    dimensions: HUD_DEFAULT_DIMENSIONS,
+                    alpha: BAR_ALPHA,
+                    localRotation: {
+                        x: 0,
+                        y: 0,
+                        z: 0,
+                        w: 1
+                    },
+                    solid: true,
+                    color: {
+                        "red": 0,
+                        "green": 255,
+                        "blue": 255
+                    },
+                    parentID: user.hudID[0]
+                },
+                {
+                    name: "Voice-HUD",
+                    drawInFront: true,
+                    shape: "Quad",
+                    localPosition: {
+                        x: 0.1833,
+                        y: 0.003,
+                        z: 0
+                    },
+                    alpha: BAR_ALPHA,
+                    dimensions: HUD_DEFAULT_DIMENSIONS,
+                    localRotation: {
+                        x: 0,
+                        y: 0,
+                        z: 0,
+                        w: 1
+                    },
+                    solid: true,
+                    color: {
+                        "red": 0,
+                        "green": 255,
+                        "blue": 255
+                    },
+                    parentID: user.hudID[0]
+                }
+            ];
+            
+            hudProperties.forEach(function(element){   
+                user.hudID.push(Overlays.addOverlay("shape", element));   
+            });
+            audioBarProperties.forEach(function(element){   
+                user.audioBarIDs.push(Overlays.addOverlay("shape", element));   
+            });
+        }, 200);
+    }
+    
+
+    // This function deletes the HUD over a user
+    function deleteHUD(uuid) {
+        var user = userStore[uuid];
+        user.hudID.forEach(function(element){
+            if (element){
```

Comment:
why is this check here?

Reply:
Was running into an error after deletion of user.hudID[0] (the parent) which could be deleting the children, leaving nothing else to find and delete.